crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -65,32 +65,35 @@ impl<S: IndexerStore> ReadApi<S> {
         descending_order: Option<bool>,
     ) -> RpcResult<TransactionsPage> {
         let limit = cap_page_limit(limit);
-        let reverse = descending_order.unwrap_or_default();
-        let indexer_seq_number = self
-            .state
-            .get_transaction_sequence_by_digest(cursor.map(|digest| digest.to_string()), reverse)?;
-        let move_call_seq_number = self
-            .state
-            .get_move_call_sequence_by_digest(cursor.map(|digest| digest.to_string()), reverse)?;
+        let is_descending = descending_order.unwrap_or_default();
+        let indexer_seq_number = self.state.get_transaction_sequence_by_digest(
+            cursor.map(|digest| digest.to_string()),
+            is_descending,
+        )?;
```

Comment:
This might be shared by multiple enum variants, but certainly not all variants, can we also put it in a if statement?

```suggestion
        let indexer_seq_number = if aaa || bbb { self.state.get_transaction_sequence_by_digest(
            cursor.map(|digest| digest.to_string()),
            is_descending,
        )?} else {None};
```